core/src/identity.rs

```diff
@@ -33,6 +33,16 @@ impl<T: crate::DIDResolver> Identity<T> {
         }
     }
 
+    pub async fn recover(
+        self,
+        did: String,
+        key_pair: Ed25519SSIKeyPair,
```

Comment:
Why not use the KeyPair trait instead of the specific one?

Reply:
KeyPair requires ed25519_zebra::SigningKey and ed25519_zebra::VerificationKey (or any other Private Public key set)

Reply:
Why does it require that? I thought we managed to design the KeyPair trait itself to be generic to such things?

Reply:
Yes but we can't do KeyPair without any specific algorithm. Our goal was to separate KeyPair from Signer, but generic KeyPair struct still has to accept the private and public key. Here is the conversion from KeyPair to signer: https://github.com/knox-networks/ssi/pull/40/files#diff-70d21711db4b5729408424f8caf1d0b5fce25c6abc89a2cc658585727e0a643eR5 and here is generic  KeyPair itself, https://github.com/knox-networks/ssi/pull/40/files#diff-1eb80c5452df3ad684ce36b4bfb4590a600978c10f0135bcba713ca18617d9bcR9 , we were implementing this on a pair code session and concluded there was no other way to have it implemented, pls correct me if I'm wrong.